### PR TITLE
[Git] Remove calls to Dir.chdir

### PR DESCRIPTION
**Pull request by DanToml**
closes https://github.com/CocoaPods/CocoaPods/issues/5880

### CODE REVIEW

File: lib/cocoapods-downloader/git.rb [hunk 1]
```diff
@@ -13,13 +13,11 @@ def options_specific?
...
-        Dir.chdir(target_path) do
-          options = {}
-          options[:git] = url
-          options[:commit] = `git rev-parse HEAD`.chomp
-          options[:submodules] = true if self.options[:submodules]
-          options
-        end
+        options = {}
+        options[:git] = url
+        options[:commit] = `git -C #{target_path} rev-parse HEAD`.chomp
```
Comment (benasher44) [inline lib/cocoapods-downloader/git.rb]: Add quotes around `#{target_path}`

**Comment by DanToml** [comment]
@benasher44 👍 updated and added a 📝 entry.

File: lib/cocoapods-downloader/git.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def options_specific?
       def checkout_options
         options = {}
         options[:git] = url
-        options[:commit] = `git -C #{target_path} rev-parse HEAD`.chomp
+        options[:commit] = `git -C "#{target_path}" rev-parse HEAD`.chomp
```
Comment (segiddins) [inline lib/cocoapods-downloader/git.rb]: why not use `git!("-C", target_path, "rev-parse", "HEAD")` ?

Comment (benasher44) [inline lib/cocoapods-downloader/git.rb]: sg2m

**Comment by DanToml** [comment]
@segiddins @benasher44 Updated

**Pull request merged**